src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/Main.java

```diff
@@ -423,21 +434,35 @@ private void execute(String mainClassName, String[] appArgs, Context context)
         ClassLoader cl = context.getClassLoader(ClassLoader.getSystemClassLoader());
         try {
             Class<?> appClass = Class.forName(mainClassName, true, cl);
-            Method main = appClass.getDeclaredMethod("main", String[].class);
-            int PUBLIC_STATIC = Modifier.PUBLIC | Modifier.STATIC;
-            if ((main.getModifiers() & PUBLIC_STATIC) != PUBLIC_STATIC) {
+            Method main = MainMethodFinder.findMainMethod(appClass);
+            if (!PreviewFeatures.isEnabled() && (!isStatic(main) || !isPublic(main))) {
```

Comment:
It seems one can run a main method without parameters without `--enable-preview` using this codepath. That is presumably not intended.

Reply:
MainMethodFinder won't return a no arg method is without --enable-preview. The local testing here is to specialize the error massages.